sdk/cosmos/azure_data_cosmos/src/query/engine.rs

```diff
@@ -0,0 +1,85 @@
+/// Represents a request from the query pipeline for data from a specific partition key range.
+pub struct QueryRequest {
```

Comment:
Don't want a `Debug` or `SafeDebug` derive here (or elsewhere, if definitely no PII)? These are affectively models, for which we de facto derive `SafeDebug`.

Reply:
Yep, I can derive `SafeDebug`, though I may file an issue for a separate pass since I believe we don't derive `SafeDebug` in our other models today and we should.

Reply:
When you're sure a model wouldn't include PII, feel free to derive `Debug`.